File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    var answer = 0;

    for (var num = 1; num <= number; num++) {
        var count = 0;
        for (var k = 1; k <= Math.sqrt(num); k++) {
            if (num % k == 0) {
                count++;
                if (k != num / k) {
                    count++;
                }
            }
            if (count > limit) {
                answer += power;
                break;
            }
        }
        if (count <= limit) {
            answer += count;
        }
    }

    return answer;
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    
    answer = 1
    for i in range(2, number + 1):
        cnt = 0
        for j in range(1, int(i ** (1/2)) +1):
            if i % j == 0 :
                cnt += 1
                if (j ** 2 != i):
                    cnt += 1
        
        if cnt <= limit:
            answer += cnt
        else:
            answer += power
    
    return answer